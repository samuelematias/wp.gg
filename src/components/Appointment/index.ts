export {
    Appointment,
    AppointementProps,
} from './Appointment';
