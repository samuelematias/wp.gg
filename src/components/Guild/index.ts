export {
    Guild,
    GuildProps,
} from './Guild';
