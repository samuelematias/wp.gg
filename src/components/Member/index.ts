export {
    Member,
    MemberProps,
} from './Member';
