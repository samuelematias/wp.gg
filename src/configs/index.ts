export {
    COLLECTION_USERS,
    COLLECTION_APPOINTMENTS
} from './database';
