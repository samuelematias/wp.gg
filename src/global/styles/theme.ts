export const theme = {
        colors: {
                primary: '#9146FF',

                secondary100: '#0E0E0E',
                secondary90: '#0D133D',
                secondary80: '#0E1647',
                secondary85: '#171F52',
                secondary70: '#1B2565',
                secondary60: '#1B2565',
                secondary50: '#243189',
                secondary40: '#1D2766',
                secondary30: '#495BCC',

                overlay: 'rgba(0,0,0,0.7)',
                highlight: '#ABB1CC',
                heading: '#DDE3F0',
                line: '#0E0E0E',
                on: '#32BD50',
        },

        fonts: {
                title700: 'Inter_700Bold',
                title600: 'Inter_600SemiBold',
                text400: 'Inter_400Regular',
                text500: 'Inter_500Medium',
        }
};
