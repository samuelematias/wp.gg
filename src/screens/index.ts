export * from './AppointmentCreate';
export * from './AppointmentDetails';
export * from './Guilds';
export * from './Home';
export * from './SignIn';
