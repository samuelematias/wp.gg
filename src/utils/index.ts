export { categories } from './categories';
export { IMAGE_PLACEHOLDER } from './constants';
