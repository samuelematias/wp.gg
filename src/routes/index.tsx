import {
    DefaultTheme,
    NavigationContainer
} from '@react-navigation/native';

import {
    SignIn
} from '../screens';

import { useAuth } from '../hooks';

import { AppRoutes } from './app.routes';

import { theme } from '../global/styles';

export function Routes() {
    const { user } = useAuth();

    const navTheme = {
        ...DefaultTheme,
        colors: {
            ...DefaultTheme.colors,
            background: theme.colors.secondary100,
        },
    };
    return (
        <NavigationContainer
            theme={navTheme}
        >
            {user?.id ? <AppRoutes /> : <SignIn />}
        </NavigationContainer>
    )
}
