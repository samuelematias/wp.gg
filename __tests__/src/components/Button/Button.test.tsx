import 'react-native';
import React from 'react';

import {
    render,
    fireEvent,
} from '../../../../jest/helper';

import { Button } from '../../../../src/components';

import { theme } from '../../../../src/global/styles';

const onPressMock = jest.fn();

const buttonLabel = 'banana';

const buttonTestId = {
    touchable: 'button',
    label: 'button-label',
};

const renderButton = <Button label={buttonLabel} onPress={onPressMock} />;

describe('Button', () => {

    // to clean the onPressMock
    // after each test.
    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render correctly and match the snapshot', () => {
        const { toJSON } = render(renderButton);

        expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly', () => {
        const { getByRole } = render(renderButton);

        const button = getByRole('button');

        expect(button).toBeTruthy();
    });

    it('should call onPress', () => {
        const { getByText } = render(renderButton);

        fireEvent.press(getByText(buttonLabel));

        expect(onPressMock).toHaveBeenCalled();
    });

    it('should call onPress 3x', () => {
        const { getByText } = render(renderButton);

        fireEvent.press(getByText(buttonLabel));
        fireEvent.press(getByText(buttonLabel));
        fireEvent.press(getByText(buttonLabel));

        expect(onPressMock).toHaveBeenCalledTimes(3);
    });

    it('should render with the color, theme.colors.primary, when the button is enabled', () => {
        const { getByTestId } = render(renderButton);

        const button = getByTestId(buttonTestId.touchable);

        expect(button.parent?.props.style.backgroundColor).toBe(theme.colors.primary);
    });

    it('should render disabled', () => {
        const { getByTestId } = render(
            <Button
                onPress={onPressMock}
                label={buttonLabel}
                disabled
            />,
        );

        const button = getByTestId(buttonTestId.touchable);

        expect(button.props).toHaveProperty('accessibilityState', { "disabled": true });
    });

    it('should render with the color, theme.colors.highlight, when the button is disabled', () => {
        const { getByTestId } = render(
            <Button
                onPress={onPressMock}
                label={buttonLabel}
                disabled
            />,
        );

        const button = getByTestId(buttonTestId.touchable);

        expect(button.parent?.props.style.backgroundColor).toBe(theme.colors.highlight);
    });

    it('should render with custom styles', () => {
        const { getByTestId } = render(
            <Button
                onPress={onPressMock}
                label={buttonLabel}
                style={{
                    backgroundColor: 'red',
                }}
            />,
        );

        const button = getByTestId(buttonTestId.touchable);

        expect(button.parent?.props.style.backgroundColor).toBe('red');
    });

    it(`should render the label, ${buttonLabel}`, () => {
        const { getByTestId } = render(renderButton);

        const button = getByTestId(buttonTestId.label);

        expect(button.props.children).toBe(buttonLabel);
    });

    it('should render the label color, theme.colors.heading', () => {
        const { getByTestId } = render(renderButton);

        const button = getByTestId(buttonTestId.label);

        expect(button.props.style[0].color).toBe(theme.colors.heading);
    });
});
