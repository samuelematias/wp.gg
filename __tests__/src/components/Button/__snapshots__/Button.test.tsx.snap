// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly and match the snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9146FF",
      "borderRadius": 8,
      "flexDirection": "row",
      "height": 56,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="Button.testId"
>
  <Text
    style={
      Array [
        Object {
          "color": "#DDE3F0",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontFamily": "Inter_500Medium",
          "fontSize": 15,
          "textAlign": "center",
        },
      ]
    }
    testID="Button.label.testId"
  >
    banana
  </Text>
</View>
`;
